### PR TITLE
Add CJS output of preview entry

### PR DESCRIPTION
**Pull request by JReinhold**
Add CJS preview entry, to be used by Jest when using Portable Stories

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "storybook-addon-kit-test-module",
```
Comment (JReinhold) [inline package.json:2]: This should never have been here.

File: package.json [hunk 2]
```diff
@@ -48,21 +48,21 @@
...
-    "@storybook/blocks": "8.0.0-beta.2",
-    "@storybook/components": "8.0.0-beta.2",
-    "@storybook/core-events": "8.0.0-beta.2",
-    "@storybook/manager": "8.0.0-beta.2",
-    "@storybook/manager-api": "8.0.0-beta.2",
-    "@storybook/preview": "8.0.0-beta.2",
-    "@storybook/preview-api": "8.0.0-beta.2",
-    "@storybook/react": "8.0.0-beta.2",
-    "@storybook/react-vite": "8.0.0-beta.2",
+    "@storybook/addon-essentials": "8.0.0",
```
Comment (yannbf) [inline package.json:51]: shouldn't the deps be ^8.0.0 instead?

File: package.json [hunk 3]
```diff
@@ -48,21 +48,21 @@
...
+    "@storybook/blocks": "8.0.0",
+    "@storybook/components": "8.0.0",
+    "@storybook/core-events": "8.0.0",
+    "@storybook/manager": "8.0.0",
+    "@storybook/manager-api": "8.0.0",
+    "@storybook/preview": "8.0.0",
+    "@storybook/preview-api": "8.0.0",
+    "@storybook/react": "8.0.0",
+    "@storybook/react-vite": "8.0.0",
     "@storybook/testing-library": "^0.2.2",
```
Comment (yannbf) [inline package.json:63]: we should get rid of this and use @storybook/test instead. Even better, if this package is not used at all, then just get rid of it entirely

File: package.json [hunk 4]
```diff
@@ -76,7 +76,7 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "rimraf": "^5.0.5",
```
Comment (yannbf) [inline package.json:78]: I think this is not needed because tsup has a clean option which is already set to true. I know this has nothing to do with the PR, they're just a small boyscout rule thing

File: package.json (hunk 2 above)
Comment (JReinhold) [reply package.json:51]: This is actually on purpose, originally discussed here https://github.com/storybookjs/addon-kit/pull/59

tl;dr by pinning to 8.0.0 (similar to what we did for 7.0.0) we're encouraging addon authors to support the whole major version range instead of just the latest minor release. If they specifically need an API from a later minor, they can upgrade to that intentionally.

Comment (yannbf) [reply package.json:49]: Got it! But doesn't that mean they won't notice issues in CI or locally? If 8.1.0 releases something that ends up affecting the addon (which we know can happen), then they won't know. This would make more sense if we set up a nightly CI check that uses npx storybook@latest upgrade and then runs yarn build and optionally yarn test-storybook

Comment (JReinhold) [reply package.json:49]: I see what you mean, it would be ideal to support "both", I don't feel like adding that CI workflow now, since I don't think it's that simple to test an addon, given they are often manager functionality which can't be represented in a story.

I'd rather optimize for addons to support a whole major range, than I'd optimize for addons to catch bugs/regression we introduce in a Storybook minor release.

**Review by yannbf — APPROVED** [review]
